### PR TITLE
Track active ssh sessions established via the web

### PR DESCRIPTION
**Pull request by rosstimothy**
Adds open session tracking in `web.Handler` and introduces a new
 `web.Server` to prevent closing the web `http.Server` during
a graceful shutdown if there are still open sessions. From the `http.Server/Shutdown` docs:

> Shutdown does not attempt to close nor wait for hijacked
> connections such as WebSockets. The caller of Shutdown should
> separately notify such long-lived connections of shutdown and wait
> for them to close, if desired.

Because the web sessions use WebSockets, `Shutdown` will not wait for them to be idle before terminating. This used to work due to the fact that web sessions dialed the Proxy SSH Port  to establish the SSH session. Which meant that all accounting of active user connections by `sshutils/server.go` was accurate. But now that sessions are established directly via the `proxy.Router` the web sessions were unaccounted for.

The `web.Server` now mimics the behavior of `Shutdown` in `sshutils/server.go`. Only once all the active web sessions have been terminated will the underlying `http.Server` be shutdown.

Fixes #20227

### CODE REVIEW

**Comment by zmb3** [comment]
FYI @rudream I wonder if this helps with the moderated sessions item we discussed today.

File: integration/helpers/instance.go
Comment (sfreiberg) [inline integration/helpers/instance.go:1523]: Consider calling this in a defer so that it doesn't leak if any of the errors between dialer.Dial() and this line are hit.

File: integration/integration_test.go
Comment (sfreiberg) [inline integration/integration_test.go:1425]: What is this TODO waiting on? Should it be done before this PR is merged?

File: lib/web/server.go
Comment (sfreiberg) [inline lib/web/server.go:32]: Comments should contain punctuation here and below.

File: lib/web/terminal.go
Comment (sfreiberg) [inline lib/web/terminal.go:894]: Is it a concern that this would replace "\r\n" with "\r\r\n"? Also since it's only used once why not replace this with strings.ReplaceAll?

**Review by sfreiberg — APPROVED** [review]

**Review by timothyb89 — APPROVED** [review]
Sorry for the delay, this seems good!

**Pull request merged**
